src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -1614,6 +1614,79 @@ class spell_gen_elune_candle : public SpellScript
     }
 };
 
+// 50051 - Ethereal Pet Aura
+enum EtherealPet
+{
+    SAY_STEAL_ESSENCE           = 1,
+    SAY_CREATE_TOKEN            = 2,
+
+    SPELL_STEAL_ESSENCE_VISUAL  = 50101,
+    SPELL_CREATE_TOKEN          = 50063
+};
+
+class spell_ethereal_pet_aura : public AuraScript
+{
+    PrepareAuraScript(spell_ethereal_pet_aura);
+
+    bool CheckProc(ProcEventInfo& eventInfo)
+    {
+        uint32 levelDiff = std::abs(GetTarget()->getLevel() - eventInfo.GetProcTarget()->getLevel());
+        if (levelDiff >= 9)
```

Comment:
change to: return levelDiff <= 9;

Reply:
shouldn't it only proc if your level diff is smaller than 10? With your propsal @Keader it won't proc anymore for diffs under 9

Reply:
what? if levelDiff <= 9, it will return true for: 0, 1, 2, 3...9

```cpp
bool CheckProc(ProcEventInfo& eventInfo)
{
    uint32 levelDiff = std::abs(GetTarget()->getLevel() - eventInfo.GetProcTarget()->getLevel());
    return levelDiff <= 9;
}
```

Reply:
ah didn't read the return xD sry :D